### PR TITLE
updates to OData feed

### PR DESCRIPTION
**Pull request by nickpell**
Product Notes: this splits the OData feed for cases by case type, includes case properties, and fixes the integration with Power BI (by removing extra fields from the feed).

🐡 

fyi @czue

### CODE REVIEW

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -961,3 +965,9 @@ def create_response(self, request, data, response_class=HttpResponse, **response
     class Meta(v0_4.CommCareCaseResource.Meta):
         resource_name = 'odata/{}'.format(ODATA_CASE_RESOURCE_NAME)
         serializer = ODataCommCareCaseSerializer()
+
+    def prepend_urls(self):
+        return [
+            url(r"^(?P<resource_name>%s)/(?P<case_type>[\w\d_.-]+)/$" % self._meta.resource_name,
+                self.wrap_view('dispatch_list'))
+        ]
```
Comment (czue) [inline corehq/apps/api/resources/v0_5.py]: neat

File: corehq/apps/api/odata/views.py [hunk 2]
```diff
@@ -19,10 +20,11 @@ def get(self, request, domain):
             '@odata.context': absolute_reverse('odata_meta', args=[domain]),
             'value': [
                 {
-                    'name': 'Cases',
+                    'name': case_type,
                     'kind': 'EntitySet',
-                    'url': 'Cases',
+                    'url': case_type,
                 }
+                for case_type in get_case_types_for_domain_es(domain)
```
Comment (czue) [inline corehq/apps/api/odata/views.py]: nice. does PowerBI actually honor the service response? I never found a way to get tableau to use it.

File: corehq/apps/api/odata/utils.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from corehq.apps.app_manager.app_schemas.case_properties import ParentCasePropertyBuilder
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.apps.reports.analytics.esaccessors import get_case_types_for_domain_es
+
+
+def get_case_type_to_properties(domain):
+    # TODO - use CaseExportDataSchema
+    apps = get_apps_in_domain(domain)
```
Comment (czue) [inline corehq/apps/api/odata/utils.py]: this is a very expensive call, no? +1 to using `CaseExportDataSchema`

File: corehq/apps/api/odata/views.py [hunk 4]
```diff
@@ -35,10 +39,13 @@ class ODataMetadataView(View):
     @method_decorator(api_auth)
     @method_decorator(toggles.ODATA.required_decorator())
     def get(self, request, domain):
-        # todo: should generate this dynamically based on the domain / case schema / data dictionary
         data_file = os.path.join(os.path.dirname(__file__), 'metadata.xml')
         with open(data_file, 'r') as f:
-            return add_odata_headers(HttpResponse(f.read(), content_type='application/xml'))
+            metadata_template = Template(f.read())
+        metadata = metadata_template.render(Context({
```
Comment (czue) [inline corehq/apps/api/odata/views.py]: Any reason you couldn't just use `render_to_string` here and not bother with doing the `Template`/`file` stuff manually?

**Review by czue — COMMENTED** [review]
nice this looks great. had a few mostly small comments. think the way the case properties are generated is the only one I'm worried about.

File: corehq/apps/api/odata/views.py (hunk 2 above)
Comment (nickpell) [reply corehq/apps/api/odata/views.py]: I'm not sure if it's useful for PowerBI or Tableau, but seemed worth including as it's part of the specification and simple to do.

File: corehq/apps/api/odata/views.py (hunk 4 above)
Comment (nickpell) [reply corehq/apps/api/odata/views.py]: https://github.com/dimagi/commcare-hq/pull/23207/commits/7262b6773b471699acf2ecdbe1ec1a6bfbb58a9e

File: corehq/apps/api/odata/utils.py (hunk 3 above)
Comment (nickpell) [reply corehq/apps/api/odata/utils.py]: https://github.com/dimagi/commcare-hq/pull/23207/commits/b117be087c65ee3b36288288d3745e6d65971414

**Comment by nickpell** [comment]
@czue Updated

**Review by millerdev — COMMENTED** [review]
Seems like @czue has done a pretty thorough review and has more context than I do here.

**Review by czue — APPROVED** [review]
woop! looking good.

**Pull request merged**